salt/states/archive.py

```diff
@@ -1387,6 +1419,104 @@ def extracted(name,
             ret['comment'] = exc.strerror
             return ret
 
+    def _file_lists(src_path, dest_path):
```

Comment:
This seems like it was pulled wholesale from the `roots` fileserver. I don't like the idea of duplicating code like this. Assuming that this approach is necessary, we should create a shared function in `salt/utils/files.py` that both salt/fileserver/roots.py and this function can both use.

Reply:
It seems that they have much more differences